File: server/recorder/src/main/java/org/rliz/cfm/release/model/ReleaseGroup.java
package org.rliz.cfm.release.model;

import org.rliz.cfm.artist.model.Artist;
import org.rliz.cfm.common.model.AbstractEntity;

import javax.persistence.*;
import java.util.Set;
import java.util.UUID;

/**
 * Represents a group of releases, i.e. a release family.
 */
@Entity
public class ReleaseGroup extends AbstractEntity {

    private UUID mbid;

    @Column(length = 511, nullable = false)
    private String title;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(foreignKey = @ForeignKey(name = "FK_releasegroup_artist"),
        inverseForeignKey = @ForeignKey(name = "FK_artist_releasegroup"))
    private Set<Artist> artists;


    /**
     * Default constructor.
     */
    public ReleaseGroup() {
        // for JPA
    }

    /**
     * Field setting constructor.
     *
     * @param mbid    the musicbrainz ID of this release group
     * @param title   the title of the release group
     * @param artists the artists of this release group
     */
    public ReleaseGroup(UUID mbid, String title, Set<Artist> artists) {
        this.mbid = mbid;
        this.title = title;
        this.artists = artists;
    }

    public UUID getMbid() {
        return mbid;
    }

    public void setMbid(UUID mbid) {
        this.mbid = mbid;
    }

    public Set<Artist> getArtists() {
        return artists;
    }

    public void setArtists(Set<Artist> artists) {
        this.artists = artists;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: server/recorder/src/main/kotlin/org/rliz/cfm/recorder/relgroup/data/ReleaseGroup.kt
package org.rliz.cfm.recorder.relgroup.data

import org.rliz.cfm.recorder.artist.data.Artist
import org.rliz.cfm.recorder.common.data.AbstractModel
import java.util.*
import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
class ReleaseGroup : AbstractModel {

    @NotNull
    @Column(length = 511, nullable = false)
    var title: String? = null

    var lastUpdated: Long? = null

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(foreignKey = ForeignKey(name = "FK_ReleaseGroup_Artists"),
            inverseForeignKey = ForeignKey(name = "FK_Artists_ReleaseGroup"))
    var artists: Set<Artist>? = null

    constructor() : super()

    constructor(uuid: UUID, title: String, lastUpdated: Long?, artists: List<Artist>) : super(uuid) {
        this.title = title
        this.lastUpdated = lastUpdated
        this.artists = artists.toSet()
    }

}